odbc-api/src/handles/connection.rs

```diff
@@ -327,4 +333,141 @@ impl<'c> Connection<'c> {
+        let null_i16 = BufferDescription {
+            kind: BufferKind::I16,
+            nullable: true,
+        };
+
+        let not_null_i16 = BufferDescription {
+            kind: BufferKind::I16,
+            nullable: false,
+        };
+
+        let null_i32 = BufferDescription {
+            kind: BufferKind::I32,
+            nullable: true,
+        };
+
+        // The definitions for these descriptions are taken from the documentation of `SQLColumns`
+        // located at https://docs.microsoft.com/en-us/sql/odbc/reference/syntax/sqlcolumns-function
+        let catalog_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_catalog_name_len()? as usize,
+            },
+            nullable: true,
+        };
+
+        let schema_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_schema_name_len()? as usize,
+            },
+            nullable: true,
+        };
+
+        let table_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_table_name_len()? as usize,
+            },
+            nullable: false,
+        };
+
+        let column_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: self.max_column_name_len()? as usize,
+            },
+            nullable: false,
+        };
+
+        let data_type_desc = not_null_i16;
+
+        const MAX_EXPECTED_TYPE_NAME_LEN: usize = 255;
+        let type_name_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_TYPE_NAME_LEN,
+            },
+            nullable: false,
+        };
+
+        let column_size_desc = null_i32;
+        let buffer_len_desc = null_i32;
+        let decimal_digits_desc = null_i16;
+        let precision_radix_desc = null_i16;
+        let nullable_desc = not_null_i16;
+
+        const MAX_EXPECTED_REMARKS_LEN: usize = 255;
+        let remarks_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_REMARKS_LEN,
+            },
+            nullable: true,
+        };
+
+        const MAX_EXPECTED_COLUMN_DEFAULT_LEN: usize = 255;
+        let column_default_desc = BufferDescription {
+            kind: BufferKind::Text {
+                max_str_len: MAX_EXPECTED_COLUMN_DEFAULT_LEN,
+            },
+            nullable: true,
+        };
+
+        let sql_data_type_desc = not_null_i16;
+        let sql_datetime_sub_desc = null_i16;
+        let char_octet_len_desc = null_i32;
+        let ordinal_pos_desc = BufferDescription {
+            kind: BufferKind::I32,
+            nullable: false,
+        };
+
+        const MAX_EXPECTED_IS_NULLABLE_LEN: usize = 10;
+        let is_nullable_desc = BufferDescription {
+            kind: BufferKind::Text {
```

Comment:
Is nullable is really a text and not a bool? I am suprised

Reply:
It's either `"YES"`, `"NO"` or `""` (zero-length string) as described under the row for `IS_NULLABLE`

Reply:
In that case we could hardcode this to `3` right?

Reply:
Yeah originally I was being extra conservative for use in the higher level wrapper (e.g. in case a driver misbehaved and returned something else), but it should be fine to use 3 here. Updated the branch